spec/models/user_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { User.new(name: 'Yusuf') }
+
+  before { subject.save }
+
+  describe 'validation tests' do
+    it 'name should be present' do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'posts_counter should be integer' do
+      subject.posts_counter = 'hey'
+      expect(subject).to_not be_valid
+    end
+
+    it 'posts_counter should be greater than or equal to zero' do
+      subject.posts_counter = -2
+      expect(subject).to_not be_valid
+      subject.posts_counter = 0
+      expect(subject).to be_valid
+    end
+  end
+
+  describe '#recent_posts' do
+    it 'returns the 3 most recent posts' do
+      # Arrange
+      user = User.create(name: 'Otmane')
+      post1 = Post.create(title: 'post1', author: user, created_at: 4.days.ago)
+      post2 = Post.create(title: 'post2', author: user, created_at: 3.days.ago)
+      post3 = Post.create(title: 'post3', author: user, created_at: 2.days.ago)
+
+      # Act
+      rec_posts = user.recent_posts
+
+      # Assert
+      expect(rec_posts).to eq([post3, post2, post1])
+    end
+  end
```

Comment:
- [ ] all test seems to be failing from my side, kindly fix
![image](https://github.com/kayjeee/Kayjeee_Postgresql_blogapp/assets/66097371/8a7daa95-f9ac-44a6-85c1-f9cbb25011ff)